src/Api/Parser/AbstractRestParser.php

```diff
@@ -93,6 +93,12 @@ private function extractHeader(
     ) {
         $value = $response->getHeaderLine($shape['locationName'] ?: $name);
 
+        if (!empty($shape->offsetGet('jsonvalue'))
```

Comment:
Any reason why not use `if($shape->offsetGet('jsonvalue'))` directly?  [`offsetGet`](https://github.com/aws/aws-sdk-php/blob/33e8f7083fcb54fd926a8569cb81a0795e8c2633/src/Api/AbstractModel.php#L30) returns value when the key is set or it will returns null, php `if` takes care of these.